src/main/downloadsManager.ts

```diff
@@ -635,7 +635,15 @@ export class DownloadsManager extends JsonFileManager<DownloadedItems> {
 
         let thumbnailData;
         if (state === 'completed' && item.getMimeType().toLowerCase().startsWith('image/')) {
-            thumbnailData = (await nativeImage.createThumbnailFromPath(overridePath ?? item.getSavePath(), {height: 32, width: 32})).toDataURL();
+            // Linux doesn't support the thumbnail creation so we have to use the base function
+            if (process.platform === 'linux') {
+                // We also will cap this at 1MB so as to not inflate the memory usage of the downloads dropdown
+                if (item.getReceivedBytes() < 1000000) {
```

Comment:
please clarify
> to not inflate the memory usage of the downloads 

Does it mean if image size is greater than 1mb we wont show the preview in the download dropdown

Reply:
Correct. This is a bit of pre-emptive safety, but I did notice that if you do have a really large image stored in the JSON (like I had before) it will slow down the app significantly, so I want to avoid that.

Reply:
Since this is for images only dont we have access to preview url of the image?

Reply:
Not anymore, we can't use the `file:///` protocol which would facilitate that. And we don't want to open up `mattermost-desktop:` to arbitrary files, or else security becomes a concern.

I'd be up for trying to figure out a better solution in the future, but this works for now.